### ISSUE TITLE
UI architecture search locator fails if its length is > 32.

### ISSUE THREAD

**Issue by sthirugn** [SatelliteQE/robottelo#2909]
![screenshot](https://cloud.githubusercontent.com/assets/3309189/10584298/73919206-765f-11e5-8a01-fbce7965da71.png)

**Comment by sthirugn** [comment]
If the text length is greater than 32, UI truncates it and adds other dom attributes to it.  To fix this, the locator should be 
```py
    "arch.arch_name": (
        By.XPATH,
        "//a/span[contains(@data-original-title,'%s') or contains(., '%s')]"
    ),
```
instead of
```py
     "arch.arch_name": (By.XPATH, "//a[contains(., '%s')]"),
```

In order to implement this, we need to pass '%s' twice and currently, the framework supports sending of one string only.

**Comment by sthirugn** [comment]
Assigning this to @oshtaier to fix this issue.

**Comment by omaciel** [comment]
@sthirugn does the xpath need to change when the length of the architecture's name is less that 32 characters or will the xpath you proposed catch both scenarios? @oshtaier could you fix this for us?

**Comment by sthirugn** [comment]
@omaciel with manual test, it looks like my fix will cover all scenarios, this is a matter of getting this in to the framework.

**Comment by oshtaier** [comment]
It is well-known issue that is covered in another entities. Here is an example of implementation for 'environment' entity, which can be applied for architecture too:
```
    def search(self, name):
        """Searches existing env from UI"""
        Navigator(self.browser).go_to_environments()
        if len(name) <= 30:
            element = self.search_entity(name, locators['env.env_name'])
        else:
            element = self.search_entity(
                name, common_locators['select_filtered_entity'])
        return element
```
If you like another solution to be implemented, please create separate issue that should be applied for all entities, not only architecture

**Comment by omaciel** [comment]
@oshtaier I think you were to quick to close this issue. Searching has caused many failures in the automation and if this is a well known issue then we would like to hear it from you and discuss it. It is only after we have a plan of attack that this issue should be closed, assuming that the fix cannot be tracked here. Reopening it.

**Comment by oshtaier** [comment]
Sure :) But used solution described above

**Comment by omaciel** [comment]
Will have a better look once I'm in front of the computer but I think we should get together and discuss this so we can hopefully have a solid solution

**Comment by oshtaier** [comment]
Fine, but as I mentioned above, if we like to have another solution, so current issue doesn't make sense and need to create one for general approach

**Comment by omaciel** [comment]
Let's discuss this today and see if we can come up with a generic solution.

**Comment by oshtaier** [comment]
Closing current issue and adding https://github.com/SatelliteQE/robottelo/issues/2913 for generic case investigation. That, of course, does not involve, initial https://github.com/SatelliteQE/robottelo/pull/2874 PR that can be closed without any improvements to existing framework

**Comment by sthirugn** [comment]
@oshtaier:
When you say `That, of course, does not involve, initial #2874 PR that can be closed without any improvements to existing framework`
- Please explain why you are proposing to close my PR #2874 and not merge it?

**Comment by oshtaier** [comment]
`merged`
